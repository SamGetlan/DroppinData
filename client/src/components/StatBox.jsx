import React from 'react';
import { Route, Link } from 'react-router-dom';
import StatGrid from './StatGrid.jsx';
import DeathMap from './DeathMap.jsx';

class StatBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      placeInput: null,
      killsInput: null,
      lootInput: null,
      gameType: null,
    }
    this.handlePlaceInputChange = this.handlePlaceInputChange.bind(this);
    this.handleKillsInputChange = this.handleKillsInputChange.bind(this);
    this.handleLootInputChange = this.handleLootInputChange.bind(this);
    this.handleGameTypeOptionChange = this.handleGameTypeOptionChange.bind(this);
    this.handleNotReadySubmit = this.handleNotReadySubmit.bind(this);
  }

  handlePlaceInputChange(e) {
    this.setState({
      placeInput: e.target.value,
    });
  }

  handleKillsInputChange(e) {
    this.setState({
      killsInput: e.target.value,
    });
  }

  handleLootInputChange(e) {
    this.setState({
      lootInput: e.target.value,
    });
  }

  handleGameTypeOptionChange(e) {
    this.setState({
      gameType: e.target.value,
    });
  }

  handleNotReadySubmit() {
    if (this.state.placeInput < 1 || this.state.placeInput > 100 || this.state.placeInput === null) {
      console.log('Your place needs to be between 1 and 100');
    } else if (this.state.killsInput < 0 || this.state.killsInput > 99 || this.state.killsInput === null) {
      console.log('Your kills need to be between 0 and 99');
    } else if (this.state.lootInput < 0 || this.state.lootInput > 10 || this.state.lootInput === null) {
      console.log('Your loot rating needs to be between 0 and 10');
    } else if (this.state.gameType !== 'solo' && this.state.gameType !== 'duo' && this.state.gameType !== 'squad') {
      console.log('please select a game type');
    }
  }

  render() {
    return (
      <div className="statBox">
        <div className="temporaryGameTypeSelector">
          <span className={this.props.currentGameType === 'solo' ? 'selected gameTypeSelector' : 'gameTypeSelector'} onClick={() => this.props.changeGameType('solo')}>Solo</span><span className={this.props.currentGameType === 'duo' ? 'selected gameTypeSelector' : 'gameTypeSelector'} onClick={() => this.props.changeGameType('duo')} >Duo</span><span className={this.props.currentGameType === 'squad' ? 'selected gameTypeSelector' : 'gameTypeSelector'} onClick={() => this.props.changeGameType('squad')} >Squad</span><span className={this.props.currentGameType === 'all' ? 'selected gameTypeSelector' : 'gameTypeSelector'} onClick={() => this.props.changeGameType('all')} >All</span>
        </div>
        <StatGrid stats={this.props.userGameData[this.props.location.name] ? this.props.userGameData[this.props.location.name][this.props.currentGameType] : undefined} />
        <div className="entryContainer">
          <h3 className="inputLabel">What place did you come in?</h3> <input type="number" id="placeInput" onChange={(e) => this.handlePlaceInputChange(e)} />
        </div>
        <div className="entryContainer">
          <h3 className="inputLabel">How many kills did you have?</h3> <input type="number" id="killsInput" onChange={(e) => this.handleKillsInputChange(e)} />
        </div>
        <div className="entryContainer">
          <h3 className="inputLabel">Rate your loot 1-10</h3> <input type="number" id="lootInput" onChange={(e) => this.handleLootInputChange(e)} />
        </div>
        {/* <div className="entryContainer">
          <h3 className="inputLabel">Where did you die?</h3> 
          <select id="deathInput" onChange={this.handleChange}>
            <option value="null"></option>
            {this.props.locations.map(location => <option value={location.camelCase}>{location.name}</option>)}
            <option value="betweenLocations">Between Locations</option>
            <option value="winner">I Won!</option>
          </select>
        </div> */}
        <div id="gameTypeOptions">
          <input type="radio" value="solo" id="soloRadio" name="gameType" checked={this.state.gameType === 'solo'} onChange={this.handleGameTypeOptionChange} /><label htmlFor="soloRadio">Solo</label>
          <input type="radio" value="duo" id="duoRadio" name="gameType" checked={this.state.gameType === 'duo'} onChange={this.handleGameTypeOptionChange} /><label htmlFor="duoRadio">Duo</label>
          <input type="radio" value="squad" id="squadRadio" name="gameType" checked={this.state.gameType === 'squad'} onChange={this.handleGameTypeOptionChange} /><label htmlFor="squadRadio">Squad</label>
        </div>
        {this.state.placeInput <= 100 && this.state.placeInput >= 1 && this.state.killsInput >= 0 && this.state.killsInput <= 99 && this.state.lootInput >= 0 && this.state.lootInput <= 10 && (this.state.gameType === 'solo' || this.state.gameType === 'duo' || this.state.gameType === 'squad') &&
        <Link className="submitGameButtonContainer" to="/home/deathLocation">
          <button className={`submitGameButton ${this.props.submitButtonState ? 'enabledButton' : 'disabledButton'}`} onClick={this.props.toggleDeathMap}>Submit Game</button>
        </Link>}
        {(this.state.placeInput <= 100 && this.state.placeInput >= 1 && this.state.killsInput >= 0 && this.state.killsInput <= 99 && this.state.lootInput >= 0 && this.state.lootInput <= 10 && (this.state.gameType === 'solo' || this.state.gameType === 'duo' || this.state.gameType === 'squad')) || 
        <div className="submitGameButtonContainer">
          <button className="submitGameButton disabledButton" onClick={this.handleNotReadySubmit} >Submit Game</button>
        </div>}
        <Route path="/home/deathLocation" render={props => <DeathMap {...props}
          submitButtonState={this.props.submitButtonState}
          handleCoordinateChoiceClick={this.props.handleDeathCoordinateChoiceClick}
          userSettings={this.props.userSettings}
          deathMapMarker={this.props.deathMapMarker}
          deathMapMarkerStyle={this.props.deathMapMarkerStyle}
          submitGame={this.props.submitButtonState ? () => { this.props.handleSubmit(this.state.placeInput, this.state.killsInput, this.state.lootInput, this.state.gameType); } : console.log('Button is temporary disabled')}
        />} />
      </div>
    );
  }
}

export default StatBox;
