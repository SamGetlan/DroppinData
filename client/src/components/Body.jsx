import React from 'react';
import ActionButton from './ActionButton.jsx';
import BackSplash from './BackSplash.jsx';
import ActiveTile from './ActiveTile.jsx';
import StatBox from './StatBox.jsx';



class Body extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      gameType: 'all',
      coordinateChoice: '',
    };
    this.changeGameType = this.changeGameType.bind(this);
  }


  changeGameType(gameType) {
    this.setState({
      gameType,
    });
  }

  checkActiveIsOk(active) {
    return this.props.filteredLocations.reduce((acc, curr) => {
      if (curr.camelCase === active) {
        acc = true;
      }
      return acc;
    }, false);
  }


  render() {
    return (
      <div>
        <ActionButton resetMarker={this.props.resetMarker} handleActionClick={this.props.handleActionClick} />
        {(this.props.active !== false && this.checkActiveIsOk(this.props.active)) &&
          <div className="activeTileContainer">
            <ActiveTile mapMarkerStyle={this.props.mapMarkerStyle} mapMarker={this.props.mapMarker} location={this.props.filteredLocations[this.props.activeIndex]} handleCoordinateChoiceClick={this.props.handleCoordinateChoiceClick}/>
            <StatBox 
              currentGameType={this.state.gameType} 
              changeGameType={this.changeGameType} 
              location={this.props.filteredLocations[this.props.activeIndex]} 
              handleSubmit={this.props.handleSubmit} 
              submitButtonState={this.props.submitButtonState} 
              locations={this.props.locations} 
              userGameData={this.props.userGameData}
              deathMapMarker={this.props.deathMapMarker}
              deathMapMarkerStyle={this.props.deathMapMarkerStyle}
              userSettings={this.props.userSettings}
              handleDeathCoordinateChoiceClick={this.props.handleDeathCoordinateChoiceClick}
              checkDeathMarkerLocation={this.props.checkDeathMarkerLocation}
              loggedIn={this.props.loggedIn}
            />
          </div>
        }
        <BackSplash 
          handleTileClick={this.props.handleTileClick} 
          resetMarker={this.props.resetMarker}
          userGames={this.props.userGames} 
          filteredLocations={this.props.filteredLocations} 
          userGameData={this.props.userGameData}
        />
      </div>
    );
  }
}

export default Body;
