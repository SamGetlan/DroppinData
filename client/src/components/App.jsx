import React from 'react';
import axios from 'axios';
import Navbar from './Navbar.jsx';
import Body from './Body.jsx';
import UserForm from './UserForm.jsx';
import FilterLocations from './FilterLocations.jsx';
import AccountOptionsForm from './AccountOptionsForm.jsx';
import FullMap from './FullMap.jsx';
import locations from '../data.js';


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      focus: '',
      locations,
      filteredLocations: locations.slice(),
      userFormActive: true,
      filterLocationsActive: false,
      signUpForm: false,
      loggedIn: false,
      active: false,
      activeIndex: false,
      userGames: null,
      submitButtonState: true,
      accountOptionsForm: false,
      logInFailed: null,
      userGameData: null,
      showFullMap: false,
    };
    this.handleUserFormClick = this.handleUserFormClick.bind(this);
    this.handleFilterClick = this.handleFilterClick.bind(this);
    this.loginUserFormOption = this.loginUserFormOption.bind(this);
    this.signUpUserFormOption = this.signUpUserFormOption.bind(this);
    this.handleAccountSignUp = this.handleAccountSignUp.bind(this);
    this.handleAccountSignIn = this.handleAccountSignIn.bind(this);
    this.filterOut = this.filterOut.bind(this);
    this.filterIn = this.filterIn.bind(this);
    this.handleGameSubmit = this.handleGameSubmit.bind(this);
    this.handleActionClick = this.handleActionClick.bind(this);
    this.handleTileClick = this.handleTileClick.bind(this);
    this.handleAccountOptionsClick = this.handleAccountOptionsClick.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
    this.getUserGameData = this.getUserGameData.bind(this);
    this.hardGroupClick = this.hardGroupClick.bind(this);
    this.notRecentGroupClick = this.notRecentGroupClick.bind(this);
    this.killsGroupClick = this.killsGroupClick.bind(this);
    this.placeGroupClick = this.placeGroupClick.bind(this);
    this.popularGroupClick = this.popularGroupClick.bind(this);
    this.filterAllIn = this.filterAllIn.bind(this);
    this.filterAllOut = this.filterAllOut.bind(this);
    this.handleShowMapClick = this.handleShowMapClick.bind(this);
    this.handleMapChoiceClick = this.handleMapChoiceClick.bind(this);
  }

  handleUserFormClick() {
    this.setState({
      userFormActive: !this.state.userFormActive,
    });
  }

  handleFilterClick() {
    this.setState({
      filterLocationsActive: !this.state.filterLocationsActive,
    });
  }

  handleActionClick() {
    const { filteredLocations } = this.state;
    const index = Math.floor(Math.random() * filteredLocations.length);
    console.log(filteredLocations[index].camelCase);
    this.setState({
      active: filteredLocations[index].camelCase,
      activeIndex: index,
    });
  }

  loginUserFormOption() {
    this.setState({
      signUpForm: false,
    });
  }

  signUpUserFormOption() {
    this.setState({
      signUpForm: true,
    });
  }

  handleAccountSignUp(username, password) {
    const context = this;
    axios.post('/api/signup', {
      username,
      password,
      createdAt: new Date(),
    })
      .then((data) => {
        console.log('signUp successful:', data.config.data);
        const user = JSON.parse(data.config.data).username;
        context.setState({
          loggedIn: user,
          userFormActive: false,
        });
      })
      .catch((err) => {
        console.log('There was an error:', err);
      });
  }

  handleAccountSignIn(username, password) {
    const context = this;
    axios.post('/api/login', {
      username,
      password,
    })
      .then((data) => {
        if (data.data.username) {
          console.log('Login successful:', data);
          const user = (data.data.username);
          context.setState({
            loggedIn: user,
            userFormActive: false,
            logInFailed: false,
          });
          axios.get('/api/games')
            .then((data) => {
              console.log('data received -->', data);
              const userGameData = context.getUserGameData(data.data);
              context.setState({
                userGames: data.data,
                userGameData,
              });
            });
        } else {
          console.log('Login failed:', data.data);
          context.setState({
            logInFailed: true,
          });
        }
      })
      .catch((err) => {
        console.log('There was an error:', err);
      });
  }

  filterIn(location) {
    console.log(this.state.filteredLocations);
    if (this.state.filteredLocations.indexOf(location) === -1) {
      this.state.filteredLocations.push(location);
      const filteredLocations = this.state.filteredLocations;
      this.setState({ filteredLocations });
    }
  }

  filterOut(location) {
    console.log(this.state.filteredLocations);
    const index = this.state.filteredLocations.indexOf(location);
    if (index !== -1) {
      this.state.filteredLocations.splice(index, 1);
      const filteredLocations = this.state.filteredLocations;
      this.setState({ filteredLocations });
    }
  }

  handleGameSubmit(place, kills, loot, gameType, death) {
    const context = this;
    this.setState({
      submitButtonState: false,
    })
    // console.log('Inside the handleSubmit function');
    // console.log(`place --> ${place}`);
    // console.log(`kills --> ${kills}`);
    // console.log('loot -->', loot);
    // console.log('death -->', death);
    // console.log(`Game Type --> ${gameType}`);
    if (place >= 1 && place <= 100 && kills <= 100 && kills >= 0 && gameType && this.state.loggedIn && death !== 'null') {
      axios.post('/api/games', {
        user: this.state.loggedIn,
        date: new Date(),
        location: this.state.filteredLocations[this.state.activeIndex].name,
        place,
        kills,
        loot,
        gameType,
        death,
      })
        .then((data) => {
          console.log('We have received Data -->', data);
          context.setState({
            userGames: data.data,
          });
          const $confirm = document.createElement('p');
          $confirm.innerHTML = 'Game was successfully submitted';
          document.getElementsByClassName('statBox')[0].appendChild($confirm);
          setTimeout(() => {
            context.setState({
              submitButtonState: true,
            })
          }, 1000);
          setTimeout(() => {
            $confirm.remove();
          }, 4000);
        })
        .catch((err) => {
          console.log('There was an Error -->', err);
        });
    } else {
      let suggestion;
      if (place < 1 || place > 100 || place === undefined) {
        suggestion = 'Your place needs to be between 1 and 100.';
      } else if (kills < 0 || kills > 100 || kills === undefined) {
        suggestion = 'Your kills need to be between 0 and 100.';
      } else if (loot < 1 || loot > 10) {
        suggestion = 'Loot rating must be between 1 and 10.';
      } else if (!gameType) {
        suggestion = 'A game type must be selected.';
      } else if (death === 'null') {
        suggestion = 'A death location must be selected.';
      } else if (!this.state.loggedIn) {
        suggestion = 'You must log in to submit a game.';
      }
      const $denied = document.createElement('p');
      $denied.innerHTML = `There was an error submitting your game. ${suggestion}`;
      document.getElementsByClassName('statBox')[0].appendChild($denied);
      setTimeout(() => {
        context.setState({
          submitButtonState: true,
        })
      }, 1000);
      setTimeout(() => {
        $denied.remove();
      }, 4000);
    }
  }

  handleTileClick(location) {
    console.log('inside handleTileClick:', location);
    const index = this.state.filteredLocations.findIndex(element => (element.camelCase === location.camelCase));
    console.log('active Location -->', location);
    console.log('active index -->', index);
    this.setState({
      active: location.camelCase,
      activeIndex: index,
    });
    document.body.scrollTop = 0; // For Safari
    document.documentElement.scrollTop = 0; // For everything else
  }

  handleAccountOptionsClick() {
    console.log('inside handleAccountOptionsClick');
    this.setState({
      accountOptionsForm: !this.state.accountOptionsForm,
    });
  }

  handleLogout() {
    const context = this;
    axios.get('/api/logout')
      .then((data) => {
        context.setState({
          loggedIn: false,
          accountOptionsForm: false,
          userGames: null,
        })
      })
      .catch((err) => {
        console.log('err:', err);
      })
  }

  getUserGameData(games) {
    const results = {};
    const getPlace = (game) => {
      if (game.gameType === 'solo') {
        return game.place;
      } else if (game.gameType === 'duo') {
        return game.place * 2;
      } else if (game.gameType === 'squad') {
        return game.place * 4;
      }
    }
    const updateStats = (results, game, gameType) => {
      results[game.location][gameType].totalKills += game.kills;
      results[game.location][gameType].totalPlace += (gameType === 'all' ? getPlace(game) : game.place);
      results[game.location][gameType].totalLoot += game.loot;
      results[game.location][gameType].bestKills = Math.max(results[game.location][gameType].bestKills, game.kills);
      results[game.location][gameType].bestPlace = Math.min(results[game.location][gameType].bestPlace, (gameType === 'all' ? getPlace(game) : game.place));
      results[game.location][gameType].bestLoot = Math.max(results[game.location][gameType].bestLoot, game.loot);
      results[game.location][gameType].recentKills = game.kills;
      results[game.location][gameType].recentPlace = (gameType === 'all' ? getPlace(game) : game.place);
      results[game.location][gameType].recentLoot = game.loot;
      results[game.location][gameType].totalGames++;
      if (game.date > results[game.location][gameType].mostRecent) {
        results[game.location][gameType].mostRecent = game.date;
      }
    }
    games.forEach((game, index, games) => {
      if (!results[game.location]) {
        results[game.location] = {
          solo: {
            averageKills: 0,
            averagePlace: 0,
            averageLoot: 0,
            bestKills: 0,
            bestPlace: 101,
            bestLoot: 0,
            totalKills: 0,
            totalPlace: 0,
            totalLoot: 0,
            totalGames: 0,
            recentKills: 0,
            recentPlace: 0,
            recentLoot: 0,
            mostRecent: game.date,
          },
          duo: {
            averageKills: 0,
            averagePlace: 0,
            averageLoot: 0,
            bestKills: 0,
            bestPlace: 101,
            bestLoot: 0,
            totalKills: 0,
            totalPlace: 0,
            totalLoot: 0,
            totalGames: 0,
            recentKills: 0,
            recentPlace: 0,
            recentLoot: 0,
            mostRecent: game.date,
          },
          squad: {
            averageKills: 0,
            averagePlace: 0,
            averageLoot: 0,
            bestKills: 0,
            bestPlace: 101,
            bestLoot: 0,
            totalKills: 0,
            totalPlace: 0,
            totalLoot: 0,
            totalGames: 0,
            recentKills: 0,
            recentPlace: 0,
            recentLoot: 0,
            mostRecent: game.date,
          },
          all: {
            averageKills: 0,
            averagePlace: 0,
            averageLoot: 0,
            bestKills: 0,
            bestPlace: 101,
            bestLoot: 0,
            totalKills: 0,
            totalPlace: 0,
            totalLoot: 0,
            totalGames: 0,
            recentKills: 0,
            recentPlace: 0,
            recentLoot: 0,
            mostRecent: game.date,
          },
        }
      }
      updateStats(results, game, 'all');
      updateStats(results, game, game.gameType);
    });
    for (var location in results) {
      results[location].solo.averageKills = results[location].solo.totalKills / results[location].solo.totalGames;
      results[location].duo.averageKills = results[location].duo.totalKills / results[location].duo.totalGames;
      results[location].squad.averageKills = results[location].squad.totalKills / results[location].squad.totalGames;
      results[location].all.averageKills = results[location].all.totalKills / results[location].all.totalGames;
      results[location].solo.averagePlace = results[location].solo.totalPlace / results[location].solo.totalGames;
      results[location].duo.averagePlace = results[location].duo.totalPlace / results[location].duo.totalGames;
      results[location].squad.averagePlace = results[location].squad.totalPlace / results[location].squad.totalGames;
      results[location].all.averagePlace = results[location].all.totalPlace / results[location].all.totalGames;
    }
    return results;
  }

  hardGroupClick() {
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const hardGroup = Object.keys(this.state.userGameData);
      hardGroup.sort((a, b) => {
        return userGameData[b][type].averagePlace - userGameData[a][type].averagePlace;
      });
      for (var i = 0; i < Math.min(hardGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === hardGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });

    } else {
      console.log('Need to save at least 10 games in order for customSorting to work');
    }
  }

  notRecentGroupClick(gameType = 'all') {
    console.log('inside notRecentGroupClick');
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const notRecentGroup = Object.keys(this.state.userGameData);
      notRecentGroup.sort((a, b) => {
        return userGameData[a][type].mostRecent - userGameData[b][type].mostRecent;
      });
      for (var i = 0; i < Math.min(notRecentGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === notRecentGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });
    }
  }

  killsGroupClick() {
    console.log('inside killsGroupClick');
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const killsGroup = Object.keys(this.state.userGameData);
      killsGroup.sort((a, b) => {
        return userGameData[b][type].averageKills - userGameData[a][type].averageKills;
      });
      for (var i = 0; i < Math.min(killsGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === killsGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });
    } else {
      console.log('Need to save at least 10 games in order for customSorting to work');
    }
  }

  placeGroupClick() {
    console.log('inside placeGroupClick');
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const easyGroup = Object.keys(this.state.userGameData);
      easyGroup.sort((a, b) => {
        return userGameData[a][type].averagePlace - userGameData[b][type].averagePlace;
      });
      for (var i = 0; i < Math.min(easyGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === easyGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });

    } else {
      console.log('Need to save at least 10 games in order for customSorting to work');
    }
  }

  popularGroupClick() {
    console.log('inside popularGroupClick');
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const popularGroup = Object.keys(this.state.userGameData);
      popularGroup.sort((a, b) => {
        return userGameData[b][type].totalGames - userGameData[a][type].totalGames;
      });
      for (var i = 0; i < Math.min(popularGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === popularGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });

    } else {
      console.log('Need to save at least 10 games in order for customSorting to work');
    }
  }

  filterAllIn() {
    this.setState({
      filteredLocations: locations.slice(),
    });
  }

  filterAllOut() {
    console.log('inside filterAllOut');
    this.setState({
      filteredLocations: [],
    });
  }

  handleShowMapClick() {
    this.setState({
      showFullMap: !this.state.showFullMap,
    });
  }

  handleMapChoiceClick(e) {
    console.log('inside handleMapChoiceClick from div -->', e.target.id);
    for (var i = 0; i < locations.length; i++) {
      if (locations[i].camelCase === e.target.id) {
        this.filterIn(locations[i]);
      }
    }
    for (var k = 0; k < this.state.filteredLocations.length; k++) {
      if (this.state.filteredLocations[k].camelCase === e.target.id) {
        this.setState({
          active: e.target.id,
          activeIndex: k,
          showFullMap: false,
        });
      }
    }
  }

  render() {
    return (
      <div id="app">
        <Navbar
          navButtons={['Home', 'Map', 'Filter Locations', 'Sign Up or Login']}
          classes={['home', 'map', 'filter', 'login']}
          handleUserFormClick={this.handleUserFormClick}
          handleFilterClick={this.handleFilterClick}
          loggedIn={this.state.loggedIn}
          handleAccountOptionsClick={this.handleAccountOptionsClick}
          handleShowMapClick={this.handleShowMapClick}
        />
        <Body
          filteredLocations={this.state.filteredLocations}
          activeIndex={this.state.activeIndex}
          handleSubmit={this.handleGameSubmit}
          active={this.state.active}
          handleActionClick={this.handleActionClick}
          userGames={this.state.userGames}
          handleTileClick={this.handleTileClick}
          submitButtonState={this.state.submitButtonState}
          locations={this.state.locations}
        />
        {this.state.userFormActive &&
        <UserForm
          handleUserFormClick={this.handleUserFormClick}
          signUpForm={this.state.signUpForm}
          loginUserFormOption={this.loginUserFormOption}
          signUpUserFormOption={this.signUpUserFormOption}
          handleAccountSignIn={this.handleAccountSignIn}
          handleAccountSignUp={this.handleAccountSignUp}
          logInFailed={this.state.logInFailed}
        />}
        {this.state.filterLocationsActive &&
        <FilterLocations
          locations={this.state.locations}
          filteredLocations={this.state.filteredLocations}
          filterOut={this.filterOut}
          filterIn={this.filterIn}
          handleFilterClick={this.handleFilterClick}
          hardGroupClick={this.hardGroupClick}
          notRecentGroupClick={this.notRecentGroupClick}
          killsGroupClick={this.killsGroupClick}
          placeGroupClick={this.placeGroupClick}
          popularGroupClick={this.popularGroupClick}
          filterAllIn={this.filterAllIn}
          filterAllOut={this.filterAllOut}
        />}
        {this.state.accountOptionsForm &&
        <AccountOptionsForm
          handleAccountOptionsClick={this.handleAccountOptionsClick}
          loggedIn={this.state.loggedIn}
          handleLogout={this.handleLogout}
        />}
        {this.state.showFullMap &&
        <FullMap
          handleShowMapClick={this.handleShowMapClick}
          locations={this.state.locations}
          handleMapChoiceClick={this.handleMapChoiceClick}
        />}
      </div>
    );
  }
}

export default App;
