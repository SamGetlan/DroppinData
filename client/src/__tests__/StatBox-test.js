import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App.jsx';
import Body from '../components/Body.jsx';
import StatBox from '../components/StatBox.jsx';
import { MemoryRouter } from 'react-router';
import locations from '../data.js';

test('StatBox renders correctly', () => {
  const element = renderer.create(
    <MemoryRouter >
      <StatBox
        currentGameType={'all'} 
        changeGameType={Body.changeGameType} 
        location={locations[0]} 
        handleSubmit={App.handleSubmit} 
        submitButtonState={true} 
        locations={locations} 
        userGameData={{}}
        deathMapMarker={[41, 42]}
        deathMapMarkerStyle={{top: '50%', left: '50%'}}
        userSettings={{}}
        handleDeathCoordinateChoiceClick={App.handleDeathCoordinateChoiceClick}
        checkDeathMarkerLocation={App.checkDeathMarkerLocation}
        loggedIn={false}
        statBoxFlashText={null}
      />
    </MemoryRouter>
  );
  let tree = element.toJSON();
  expect(tree).toMatchSnapshot();
});