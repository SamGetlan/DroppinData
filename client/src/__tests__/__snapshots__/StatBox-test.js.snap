// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatBox renders correctly 1`] = `
<div
  className="statBox"
>
  <div
    className="temporaryGameTypeSelector"
  >
    <span
      className="gameTypeSelector"
      onClick={[Function]}
    >
      Solo
    </span>
    <span
      className="gameTypeSelector"
      onClick={[Function]}
    >
      Duo
    </span>
    <span
      className="gameTypeSelector"
      onClick={[Function]}
    >
      Squad
    </span>
    <span
      className="selected gameTypeSelector"
      onClick={[Function]}
    >
      All
    </span>
  </div>
  <div
    className="statGridContainer"
  >
    <div
      className="gridItem"
    />
    <div
      className="gridItem"
    >
      Kills
    </div>
    <div
      className="gridItem"
    >
      Place
    </div>
    <div
      className="gridItem"
    >
      Loot
    </div>
    <div
      className="gridItem best"
    >
      Best
    </div>
    <div
      className="gridItem best"
    >
      N/A
    </div>
    <div
      className="gridItem best"
    >
      N/A
    </div>
    <div
      className="gridItem best"
    >
      N/A
    </div>
    <div
      className="gridItem average"
    >
      Average
    </div>
    <div
      className="gridItem average"
    >
      N/A
    </div>
    <div
      className="gridItem average"
    >
      N/A
    </div>
    <div
      className="gridItem average"
    >
      N/A
    </div>
    <div
      className="gridItem prev"
    >
      Previous
    </div>
    <div
      className="gridItem prev"
    >
      N/A
    </div>
    <div
      className="gridItem prev"
    >
      N/A
    </div>
    <div
      className="gridItem prev"
    >
      N/A
    </div>
  </div>
  <div
    className="entryContainer"
  >
    <h3
      className="inputLabel"
    >
      What place did you come in?
    </h3>
     
    <input
      id="placeInput"
      onChange={[Function]}
      type="number"
    />
  </div>
  <div
    className="entryContainer"
  >
    <h3
      className="inputLabel"
    >
      How many kills did you have?
    </h3>
     
    <input
      id="killsInput"
      onChange={[Function]}
      type="number"
    />
  </div>
  <div
    className="entryContainer"
  >
    <h3
      className="inputLabel"
    >
      Rate your loot 1-10
    </h3>
     
    <input
      id="lootInput"
      onChange={[Function]}
      type="number"
    />
  </div>
  <div
    className="entryContainer"
  >
    <h3
      className="inputLabel"
    >
      Did you die from the storm?
    </h3>
    <select
      id="stormDeath"
      onChange={[Function]}
    >
      <option
        id="falseStormDeath"
        value="false"
      >
        No
      </option>
      <option
        id="trueStormDeath"
        value="true"
      >
        Yes
      </option>
    </select>
  </div>
  <div
    id="gameTypeOptions"
  >
    <input
      checked={false}
      id="soloRadio"
      name="gameType"
      onChange={[Function]}
      type="radio"
      value="solo"
    />
    <label
      htmlFor="soloRadio"
    >
      Solo
    </label>
    <input
      checked={false}
      id="duoRadio"
      name="gameType"
      onChange={[Function]}
      type="radio"
      value="duo"
    />
    <label
      htmlFor="duoRadio"
    >
      Duo
    </label>
    <input
      checked={false}
      id="squadRadio"
      name="gameType"
      onChange={[Function]}
      type="radio"
      value="squad"
    />
    <label
      htmlFor="squadRadio"
    >
      Squad
    </label>
  </div>
  <div
    className="submitGameButtonContainer"
  >
    <button
      className="submitGameButton disabledButton"
      onClick={[Function]}
    >
      Submit Game
    </button>
  </div>
</div>
`;
